lib/validators.ts

```diff
@@ -79,17 +80,11 @@ function validate(bool, cb, options) {
   }
 }
 
-class ParameterError extends Error {
-  constructor(...params) {
+export class ParameterError extends Error {
+  constructor(...params: any[]) {
     super(...params);
+    if (Object.setPrototypeOf) {
+      Object.setPrototypeOf(this, ParameterError.prototype);
+    }
```

Comment:
This shouldn't be necessary; the bug you're fixing here is an artifact of our tsconfig. We're currently targeting ES5, which predates classes. By changing to a newer target, the error disappears. (Changing to a newer target has other implications, though, so I think it's fine to leave for now. I have a follow up PR waiting where we can discuss this further.)

Reply:
i did want to ask you about that.  if we're continually targeting modern versions of node and dropping support for versions outside of the supported LTS range we should tweak some of our settings to reflect that.

Reply:
I've done the work to target node 16 in #269.